### PR TITLE
Updated packet-cloud-controller deployment

### PR DESCRIPTION
**Pull request by sltet**
- fixing the container image command
- updating the env var `PACKET_AUTH_TOKEN` to `PACKET_API_KEY`

**How to categorize this PR?**

/area TODO
/kind TODO
/priority normal

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @sltet for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

**Comment by sltet** [comment]
I did a `make generate` already but nothing changed.

**Review by timebertt — APPROVED** [review]
> I did a make generate already but nothing changed.

Yes, there shouldn't be any changes in generated files for this PR :)
/lgtm

File: charts/internal/seed-controlplane/charts/packet-cloud-controller-manager/templates/deployment.yaml [hunk 1]
```diff
@@ -35,13 +35,14 @@ spec:
         image: {{ index .Values.images "cloud-controller-manager" }}
         imagePullPolicy: IfNotPresent
         command:
-          - ./cloud-controller-manager
+          - ./packet-cloud-controller-manager
```
Comment (timebertt) [inline charts/internal/seed-controlplane/charts/packet-cloud-controller-manager/templates/deployment.yaml:38]: I assume, this wasn't working before, right?

**Review by rfranzke — APPROVED** [review]
/lgtm

File: charts/internal/seed-controlplane/charts/packet-cloud-controller-manager/templates/deployment.yaml (hunk 1 above)
Comment (timebertt) [reply charts/internal/seed-controlplane/charts/packet-cloud-controller-manager/templates/deployment.yaml:38]: Seems to be the case. Quickly checked the image:
```
$ docker run -it docker.io/packethost/packet-ccm:v1.0.0 ./cloud-controller-manager
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"./cloud-controller-manager\": stat ./cloud-controller-manager: no such file or directory": unknown.
$ docker run -it docker.io/packethost/packet-ccm:v1.0.0 ./packet-cloud-controller-manager
provider config error: environment variable "PACKET_API_KEY" is required
```

**Pull request merged**